src/bidiMapper/BidiServer.ts

```diff
@@ -38,7 +38,8 @@ export class BidiServer extends EventEmitter<BidiServerEvents> {
   private constructor(
     bidiTransport: BidiTransport,
     cdpConnection: CdpConnection,
-    selfTargetId: string
+    selfTargetId: string,
+    parser: BidiParser
```

Comment:
Did you consider changing the `BidiTransport` interface so that it operates not JSON string commands - responses, but the typed objects? It would allow to move parser into the `BidiTransport` implementation if need, or send unvalidated objects directly in case of Puppeteer. It would allow to remove params parsing from the CommandProcessor.

Reply:
BidiTransport operates on the typed objects already. It does not allow moving the parser though because of the channels :D